website/content/en/highlights/2022-05-03-0-22-0-upgrade-guide.md

```diff
@@ -166,3 +170,430 @@ With this change, the message would be encoded as:
 ```text
 lvl=info msg="{\"some\":\"val\"}"
 ```
+
+### Deprecations
+
+#### Imminent removal of deprecated transforms replaced by `remap` and `reduce` {#deprecated-transforms}
+
+When the `remap` transform was introduced, several transforms were deprecated
+and removed from the documentation, despite still being available in Vector.
+Additionally the `merge` transform was replaced by `reduce`. In 0.23.0 we will
+be finally removing these transforms as part of some cleanup to our reference
+documentation.
+
+Transforms to be removed:
+
+- `add_fields`
+- `add_tags`
+- `ansi_stripper`
+- `coercer`
+- `concat`
+- `grok_parser`
+- `json_parser`
+- `key_value_parser`
+- `logfmt_parser`
+- `merge`
+- `regex_parser`
+- `remove_fields`
+- `remove_tags`
+- `rename_fields`
+- `split`
+- `tokenizer`
+
+See below for examples of how to replicate functionality using the `remap`
+transform.
+
+##### `add_fields`
+
+Before:
+
+```toml
+[transforms.add_fields]
+type = "add_fields"
+inputs = ["some_input"]
+fields.parent.child2 = "value2"
+```
+
+After:
+
+```toml
+[transforms.add_fields]
+type = "remap"
+inputs = ["some_input"]
+source = '''
+.parent.child2 = "value2"
+'''
+```
+
+##### `add_tags`
+
+Before:
+
+```toml
+[transforms.add_tags]
+type = "add_tags"
+inputs = ["some_input"]
+tags.some_tag = "some_value"
+```
+
+After:
+
+```toml
+[transforms.add_tags]
+type = "remap"
+inputs = ["some_input"]
+source = '''
+.tags = "some_value"
+'''
+```
+
+##### `ansi_stripper`
+
+Before:
+
+```toml
+[transforms.ansi_stripper]
+type = "ansi_stripper"
+inputs = ["some_input"]
+```
+
+After:
+
+```toml
+[transforms.ansi_stripper]
+type = "remap"
+inputs = ["some_input"]
+drop_on_err = false
+source = '''
+.message = strip_ansi_escape_codes!(.message)
+'''
+```
+
+##### `coercer`
+
+Before:
+
+```toml
+[transforms.coercer]
+type = "coercer"
+inputs = ["some_input"]
+types.bool =
+types.some_bool = "bool"
+types.some_float = "float"
+types.some_int = "int"
+types.some_string = "string"
+types.some_timestamp = "timestamp"
+```
+
+After:
+
+```toml
+[transforms.coercer]
+type = "remap"
+inputs = ["some_input"]
+source = '''
+.some_bool = to_bool!(.some_bool)
+.some_float = to_float!(.some_float)
+.some_int = to_int!(.some_int)
+.some_string = to_string!(.some_string)
+.some_timestamp = to_timestamp!(.some_timestamp)
+'''
+```
```

Comment:
Maybe a small comment that users can create a local variable and see the fields of that object, and return it at the end, instead, to mimic the behavior of `drop_unspecified`?